src/fuel-vm/instruction-set.md

```diff
@@ -2269,19 +2269,19 @@ To get the address from the public key, hash the public key with [SHA-2-256](../
 
 ### `ED19`: EdDSA curve25519 verification
 
-|             |                                                                                                                                                     |
-|-------------|-----------------------------------------------------------------------------------------------------------------------------------------------------|
-| Description | Verification recovered from 32-byte public key starting at `$rA` and 64-byte signature starting at `$rB` on 32-byte message hash starting at `$rC`. |
-| Operation   | ```ed19verify(MEM[$rA, 32], MEM[$rB, 64], MEM[$rC, 32]);```                                                                                         |
-| Syntax      | `ed19 $rA, $rB, $rC`                                                                                                                                |
-| Encoding    | `0x00 rA rB rC -`                                                                                                                                   |
-| Notes       |                                                                                                                                                     |
+|             |                                                                                                                             |
+|-------------|-----------------------------------------------------------------------------------------------------------------------------|
+| Description | Verification 64-byte signature at `$rB` with 32-byte public key at `$rA` for a message starting at `$rC` with length `$rD`. |
+| Operation   | ```ed19verify(MEM[$rA, 32], MEM[$rB, 64], MEM[$rC, $rD]);```                                                                |
+| Syntax      | `ed19 $rA, $rB, $rC, $rD`                                                                                                   |
+| Encoding    | `0x00 rA rB rC rD`                                                                                                          |
+| Notes       | Takes message instead of hash. **For backwards compatibility reasons, if `$rD == 0`, it will be set treated as `32`.**      |
```

Comment:
Minor, but can we say "the value of `$rD` will default to `32`" instead of "it will be set treated as `32`"

Reply:
"set treated" was typo, definitely not good. However, saying that it "will default to 32" isn't technically correct, as zero would be a valid value as well if not for this mapping. (i.e. you can say that a missing value defaults to 32, but since the value cannot be missing, it's weird to say that). 

I changed it to be just

> if `$rD == 0`, it will be treated as `32`

But maybe something like

> `if $rD == 0` will default to `32` instead

could be used as well

Reply:
But `null` is also valid here if the dev is using an old version of the opcode, so we should say "if `$rD == 0` or is `null`, the value of `$rD` will default to `32`", or something in that realm, correct?

Reply:
There's no "null" in the instruction representation. The field is just a six-bit integer, and the old version required those bits to be zeroes.

Reply:
Ah, got it. Sounds good to me then, thanks!